hamilton/plugins/polars_extensions.py

```diff
@@ -499,6 +507,146 @@ def name(cls) -> str:
+    formulas: Dict[str, Union[str, Dict[str, str]]] = None
+    float_precision: int = 3
+    has_header: bool = True
+    autofilter: bool = True
+    autofit: bool = False
+    hidden_columns: Union[Sequence[str], SelectorType] = None
+    hide_gridlines: bool = False
+    sheet_zoom: int= None
+    freeze_panes: Union[str, Tuple[int, int], Tuple[str, int, int], Tuple[int, int, int, int]]  = None
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return [DATAFRAME_TYPE]
+
+    def _get_saving_kwargs(self):
+        kwargs = {}
+        if self.worksheet is not None:
+            kwargs["worksheet"] = self.worksheet
+        if self.position is not None:
+            kwargs["position"] = self.position
+        if self.table_style is not None:
+            kwargs["table_style"] = self.table_style
+        if self.table_name is not None:
+            kwargs["table_name"] = self.table_name
+        if self.column_formats is not None:
+            kwargs["column_formats"] = self.column_formats
+        if self.dtype_formats is not None:
+            kwargs["dtype_formats"] = self.dtype_formats
+        if self.conditional_formats is not None:
+            kwargs["conditional_formats"] = self.conditional_formats
+        if self.header_format is not None:
+            kwargs["header_format"] = self.header_format
+        if self.column_totals is not None:
+            kwargs["column_totals"] = self.column_totals
+        if self.column_widths is not None:
+            kwargs["column_widths"] = self.column_widths
+        if self.row_totals is not None:
+            kwargs["row_totals"] = self.row_totals
+        if self.row_heights is not None:
+            kwargs["row_heights"] = self.row_heights
+        if self.sparklines is not None:
+            kwargs["sparklines"] = self.sparklines
+        if self.formulas is not None:
+            kwargs["formulas"] = self.formulas
+        if self.float_precision is not None:
+            kwargs["float_precision"] = self.float_precision
+        if self.has_header is not None:
+            kwargs["has_header"] = self.has_header
+        if self.autofilter is not None:
+            kwargs["autofilter"] = self.autofilter
+        if self.autofit is not None:
+            kwargs["autofit"] = self.autofit
+        if self.hidden_columns is not None:
+            kwargs["hidden_columns"] = self.hidden_columns
+        if self.hide_gridlines is not None:
+            kwargs["hide_gridlines"] = self.hide_gridlines
+        if self.sheet_zoom is not None:
+            kwargs["sheet_zoom"] = self.sheet_zoom
+        if self.freeze_panes is not None:
+            kwargs["freeze_panes"] = self.freeze_panes
+        return kwargs
+
+    def save_data(self, data: DATAFRAME_TYPE) -> Dict[str, Any]:
+        data.write_excel(self.workbook, **self._get_saving_kwargs())
+        return utils.get_file_metadata(self.workbook)
```

Comment:
this might not work if it's a workbook object. So probably want to check the type here/pull out some information if possible.